### PR TITLE
Fix reading Selafin files with more generic approach

### LINKED ISSUE #286: [SELAFIN] unable read Fudaa-prepro selafin export
[Fudaa-prepro](https://fudaa-project.atlassian.net/wiki/spaces/PREPRO/overview) is used for pre and post processing of TELEMAC2D model.

I have an "invalid string position" when importing a mesh exported from fudaa (see attached file).
Qgis 3.14.1

[export_fudaa.zip](https://github.com/lutraconsulting/MDAL/files/5006839/export_fudaa.zip)

### PR DESCRIPTION
**Pull request by vcloarec**
fix #286 
This allow to read Selafin files created by third software.
It will be good to backport to 0.9.

Funded by https://www.isl.fr/

### CODE REVIEW

File: mdal/frmts/mdal_selafin.hpp [hunk 1]
```diff
@@ -60,7 +60,10 @@ namespace MDAL
       //! Initializes and open the file file with the \a fileName
       void initialize();
 
-      //! Extracts data from files
+      //! Read the header of the file and return the project name
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_selafin.hpp:63]: Reads

File: mdal/frmts/mdal_selafin.cpp [hunk 2]
```diff
@@ -159,10 +158,14 @@ void MDAL::SelafinFile::parseFile()
 
   /* 1 record containing table X (real array of dimension NPOIN containing the
      abscisse of the points)
+     AND it is here we can now if float is simple or double precision:
+     size of record / number of vertices gives the size of the float -> 4 : simple precision -> 8 : double precision
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_selafin.cpp:162]: Can you please rewrite the sentence, it is not (gramatically) clear what is meant

**Review by PeterPetrik — APPROVED** [review]

**Comment by PeterPetrik** [comment]
We do not need to backport, since we will release 0.9.1 from master I believe

File: mdal/frmts/mdal_selafin.cpp [hunk 3]
```diff
@@ -158,8 +158,9 @@ void MDAL::SelafinFile::parseMeshFrame()
 
   /* 1 record containing table X (real array of dimension NPOIN containing the
      abscisse of the points)
-     AND it is here we can now if float is simple or double precision:
-     size of record / number of vertices gives the size of the float -> 4 : simple precision -> 8 : double precision
+     AND here, we can know if float of this file is simple or double precision:
+     reuslt of size of record divided by number of vertices gives the byte size of the float:
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_selafin.cpp:162]: reuslt -> typo

**Comment by PeterPetrik** [comment]
Isn't it more of enhancement than bugfix? I propose to bump version in mdal.cpp to 0.9.90 a create pull request in QGIS on master and test it in the nightly builds?

**Comment by nicogodet** [comment]
From user's perspective, it's more a bug than a missing feature as TELEMAC perfectly handle SELAFINs created by these third-party software.

With all mesh enhancements done, 3.22 is a really interesting LTR release and compagnies are developing plugins around theses new features like https://github.com/Artelia/Mesh_tools which is currently developed by Artelia, ISL and BRGM.

This fix was the finishing touch to having a QGIS fully capable of handling all kinds of SELAFIN and starting to modify our workflow by making QGIS a centerpiece of it.

**Comment by vcloarec** [comment]
> Isn't it more of enhancement than bugfix?

#286 is labeled as bug

**Pull request merged**